src/NServiceBus.Core/CircuitBreakers/RepeatedFailuresOverTimeCircuitBreaker.cs

```diff
@@ -28,7 +29,7 @@ public void Success()
             Logger.InfoFormat("The circuit breaker for {0} is now disarmed", name);
         }
 
-        public void Failure(Exception exception)
+        public Task Failure(Exception exception)
```

Comment:
@SeanFeldman yeah MS released that advice under the misconception that people would be implementing both sync and async API in the same lib. turns out "both sync and async" means you need to basically need to implement the entire stack twice. as such the general advice now is to "if it is IO bout only do async and do not suffix"